samples/tfm_integration/psa_firmware/README.rst

```diff
@@ -0,0 +1,187 @@
+handle secure TF-M API calls and responses. The OS-specific code to handle
+the IPC calls is in ``tfm_ipc.c``.
+
+The sample prints test info to the console either as a single-thread or
+multi-thread application.
+
+Building and Running
+********************
+
+This project needs another firmware as the update payload. It must use another
+example's hex file, and should be specified on the command line
+as ZEPHYR_FIRMWARE_UPDATE_SAMPLE_BUILD. For example, to use the sample
+`tfm_integration/tfm_ipc` as the payload:
+
+.. code-block:: bash
+
+   cd <ZEPHYR_ROOT>
+   west build -p -b lpcxpresso5s69_ns samples/tfm_integration/psa_firmware \
+       -d build/lpcxpresso55s69_ns/tfm_integration/psa_firmware
+       -- -DCONFIG_FIRMWARE_UPDATE_IMAGE=`realpath build/lpcxpresso55s69_ns/tfm_integration/tfm_ipc/zephyr_ns_signed.hex`
+
+
+This project outputs startup status and info to the console. It can be built and
+executed on an ARM Cortex M33 target board or QEMU.
+
+This sample will only build on a Linux or macOS development system
+(not Windows), and has been tested on the following setups:
+
+- macOS Big Sur using QEMU 6.0.0 with gcc-arm-none-eabi-9-2020-q2-update
+- Linux (NixOS) using QEMU 6.2.50 with gcc from Zephyr SDK 0.13.2
+
+On MPS3 AN547:
+===============
+
+1. Build Zephyr with a non-secure configuration
+   (``-DBOARD=mps2_an521_ns``).
+
+   Using ``west``
+
+   .. code-block:: bash
+
+      cd <ZEPHYR_ROOT>
+      west build -p -b mps3_an547_ns samples/tfm_integration/psa_firmware
+
+
+   Using ``cmake`` and ``ninja``
+
+   .. code-block:: bash
+
+      cd <ZEPHYR_ROOT>/samples/tfm_integration/psa_firmware/
+      rm -rf build
+      mkdir build && cd build
+      cmake -GNinja -DBOARD=mps2_an521_ns ..
+      ninja
+
+   Using ``cmake`` and ``make``
+
+   .. code-block:: bash
+
+      cd <ZEPHYR_ROOT>/samples/tfm_integration/psa_firmware/
+      rm -rf build
+      mkdir build && cd build
+      cmake -DBOARD=mps3_an547_ns ..
+      make
+
+2. Copy application binary files (mcuboot.bin and tfm_sign.bin) to
+   ``<MPS3 device name>/SOFTWARE/``.
+
+3. Edit (e.g., with vim) the ``<MPS3 device name>/MB/HBI0263C/AN547/images.txt``
+   file, and update it as shown below:
+
+   .. code-block:: bash
+
+      TITLE: Versatile Express Images Configuration File
+
+      [IMAGES]
+      TOTALIMAGES: 2 ;Number of Images (Max: 32)
+
+      IMAGE0ADDRESS: 0x10000000
+      IMAGE0FILE: \SOFTWARE\mcuboot.bin  ; BL2 bootloader
+
+      IMAGE1ADDRESS: 0x10080000
+      IMAGE1FILE: \SOFTWARE\tfm_sign.bin ; TF-M with application binary blob
+
+4. Save the file, exit the editor, and reset the MPS3 board.
+
+On QEMU:
+========
+
+Build Zephyr with a non-secure configuration (``-DBOARD=mps3_an547_ns``)
+and run it in qemu via the ``run`` command.
+
+   Using ``west``
+
+   .. code-block:: bash
+
+      cd <ZEPHYR_ROOT>
+      west build -p -b mps3_an547_ns samples/tfm_integration/psa_firmware -t run
+
+   Using ``cmake`` and ``ninja``
```

Comment:
I think we have stopped documenting using cmake ninja/make, and only document using west commands.

Maybe we shouldn't have these?

Reply:
Happy to remove them.